lib/fennec/stun.ex

```diff
@@ -2,16 +2,21 @@ defmodule Fennec.STUN do
   @moduledoc false
   # Processing of STUN messages
 
-  alias Jerboa.Params
+  alias Fennec.TURN
 
-  @spec process_message!(binary, Fennec.ip, Fennec.portn) :: binary | no_return
-  def process_message!(data, ip, port) do
-    {:ok, x = %Params{method: :binding}} = Jerboa.Format.decode(data)
-    case Fennec.Evaluator.service(x, %{address: ip, port: port}) do
-      :void ->
-        :void
-      y when is_map(y) ->
-        Jerboa.Format.encode(y)
+  @spec process_message(binary, Fennec.ip, Fennec.portn, TURN.t) ::
+    {:ok, {binary, %TURN{}}} | {:error, term}
+  def process_message(data, ip, port, turn_state) do
```

Comment:
I think ebert allows for 2 levels of whatever block. This function will be rewritten though anyway, since I'm adding authentication here.